Tests/PlayerTests/Player/PlaybackEngine/Internal/Events/PlayerEventSenderTests.swift

```diff
@@ -116,41 +124,7 @@ extension PlayerEventSenderTests {
 
 	// MARK: StreamingSessionStart
 
-	func test_send_StreamingMetricsEvent_with_StreamingSessionStart() {
-		PlayerWorld.developmentFeatureFlagProvider.shouldSendEventsInDeinit = false
-
-		let streamingSessionStart = StreamingSessionStart.mock()
-		playerEventSender.send(streamingSessionStart)
-
-		optimizedWait {
-			!eventSender.sentEvents.isEmpty
-		}
-		assertStreamingMetricsEvent(
-			name: StreamingMetricNames.streamingSessionStart,
-			group: EventGroup.streamingMetrics,
-			payload: streamingSessionStart
-		)
-	}
-
-	func test_send_StreamingMetricsEvent_with_StreamingSessionStart_legacy() {
-		PlayerWorld.developmentFeatureFlagProvider.shouldSendEventsInDeinit = true
-
-		let streamingSessionStart = StreamingSessionStart.mock()
-		playerEventSender.send(streamingSessionStart)
-
-		optimizedWait {
-			!eventSender.sentEvents.isEmpty
-		}
-		assertStreamingMetricsEvent(
-			name: StreamingMetricNames.streamingSessionStart,
-			group: EventGroup.streamingMetrics,
-			payload: streamingSessionStart
-		)
-	}
-
-	func test_send_StreamingMetricsEvent_with_StreamingSessionStart_legacy2() {
-		PlayerWorld.developmentFeatureFlagProvider.shouldSendEventsInDeinit = true
-
+	func test_send_StreamingMetricsEvent_with_StreamingSessionStart_legacy() async {
```

Comment:
[suggestion] I will only write here but it is valid for each set of test which we assert the same. Basically i had added 4 for each set because we have to cover the 2 existing conditions which result in 4 different combinations:
- send events in deinit and use EP
- send events in deinit and don't use EP
- don't send events in deinit and use EP
- don't send events in deinit and don't use EP
Let me know if I totally missed something. 🙈

Reply:
That is correct for `EventsTests` as they are testing at the `PlayerEngine` level.

In `PlayerEventSenderTests` we are jsut creating mock objects and sending them explicitly, so it doesn't test anything about `deinit` vs no `deinit` (It's `PlayerItem.deinit`, and there are no PlayerItems here)